### ISSUE TITLE
Do you have any tips getting this to work in Home Assistant?

### ISSUE THREAD

**Issue by AdrianGarside** [slynn1324/BambuP1Streamer#2]
Reading up on go2rtc I was hopeful I might be able to display the feed in my home assistant dashboard without any transcoding (which added massive latency).

I read that go2rtc could republish the content unmodified as webrtc but when I try and view the webrtc feed from this, I get an error:
![image](https://github.com/slynn1324/BambuP1Streamer/assets/26147576/dc1b9f78-6183-4721-9281-8e5eb8eb3efe)

I also tried the rtsp feed which never loads. The mjpeg feed is working well in a browser though.

**Comment by slynn1324** [comment]
I've only tested this with a P1S - which I believe publishes a MJPEG stream only.  I don't think WebRTC actually supports "MJPEG" video, so it would have to be transcoded to view in that format.  There are some example files in the www folder that make a custom home assistant component that will work with the go2rtc custom 'web socket' jpg stream.

**Comment by AdrianGarside** [comment]
Fair enough. I was able to use the mjpeg stream semi-successfully. It still had sometimes very variable latency rendering in both web page views and the iphone app (especially) with transient errors at startup but once it got going it was reliable - but given the iphone app reloads the dashboard whenever you leave and come back to it, that was kinda annoying and sometimes/often requiring a manual refresh to load it properly in the app.

I ended up using the newly discovered data about what the port 6000 format was on P1Ps to create an image sensor in the integration that updates each time there's a new jpeg published on that port. Since it's running continuously its always instantly present and up to date.
